wasm/main.py

```diff
@@ -2811,40 +2801,39 @@ def spec_externtype_matching_limits(limits_a: Limits, limits_b: Limits) -> str:
 def spec_externtype_matching(externtype1, externtype2):
     logger.debug('spec_externtype_matching(%s, %s)', externtype1, externtype2)
 
-    if externtype1[0] != externtype2[0]:
-        raise Unlinkable(f"Mismatch in extern types: {externtype1[0]} != {externtype2[0]}")
-    elif "func" == externtype1[0] and "func" == externtype2[0]:
-        if externtype1[1] == externtype2[1]:
+    if type(externtype1) is not type(externtype2):
```

Comment:
I can't think of a reason this would be a problem, but `is` tests usually stand out to me as needing a reason. I can imagine a python implementation (like PyPy) breaking in weird ways. So `==` seems life the safer, standard bet, but it's really just intuition.

Reply:
same reason why we use `if value is None`.  Since classes are singletons I have historically used the convention of using `is` checks  for equality between classes.

Reply:
> classes are singletons

Ah, didn't realize this was a python runtime guarantee. I couldn't figure out how to google for it. The closest I could find was this [SO post](https://stackoverflow.com/a/51503803/8412986):
>  class equality tests only ever return true for objects that are identical. reference_to_class == other_reference_to_class is only ever True if both references are to the same class object; the test is essentially the same as reference_to_class is other_reference_to_class.

Good enough for me :+1: